### ISSUE TITLE
Looping text on Sheik cutscene in ToT....

### ISSUE THREAD

**Issue by Trudovah** [AmazingAmpharos/OoT-Randomizer#130]
When I first become adult Link, Sheik won't stop repeating the poem and never ends the scene... I am not sure what happened or how to fix this.... help?

**Comment by Cafrado** [comment]
I don't know for sure what causes it but i think it may be that you might need the Ocarina of time (the fairy ocarina might not work if you are an adult) and since you don't have it the text just bugs and starts looping

**Comment by AmazingAmpharos** [comment]
This is a confusing issue for me since Sheik shouldn't exist in the randomizer at all (anywhere! We killed Sheik). What version of the randomizer did this occur on and what settings did you use?

**Comment by Trudovah** [comment]
Which would be why she's screwed up! I downloaded straight from here about 1 month again or so. And using the most current release of Project64. 
I have some other crashing issues, like the menu not loading properly and crashing the game, but I can deal with that!

**Comment by Trudovah** [comment]
I'll see if I can record something...maybe that'll help?

**Comment by Trudovah** [comment]
Ok. So it won't let me add a short video compressed! Too big! So, the story is, Link gets the mastersword, the Ganon cutscene plays and then the Rauru cutscene. He returns to ToT and that cutscene starts playing as per normal, Sheik shows up and just keeps telling the story over and over again!

**Comment by TopHatMan** [comment]
This happens when you alter patch.py file and stop it from writing some flags. AA should know more about this. But this should not happen unless your altering things.

**Comment by Zelsaard** [comment]
I also had (maybe still have) this problem. Are there other cutscenes that have played out in full during your run? For example, during the run in which I encountered this problem I was also shown the full opening cutscene, while cutscenes like the fairy ocarina were properly removed. 

After reading through other issue threads I noticed that the latest version of the randomizer requires Python 3.6 or later for it to properly work. After installing the newest Python version (3.7 as of writing) and starting a new run, I see that the intro cutscene is properly removed and I have a random medallion/stone to start (which I didn't when I encountered the problem because the becoming an adult cutscene played out normally and I was granted the light medallion).

I will provide an update on this once I reach the ToT again.

**Comment by Trudovah** [comment]
I did have the full opening cutscene as well! And I haven't altered anything, aside from what the Randomizer did.

**Comment by Trudovah** [comment]
If you want, I could email you or make available to rom I am using, so you have the exact code I am running.

**Comment by TopHatMan** [comment]
So you did no edits to any py file and this occurred?

**Comment by Trudovah** [comment]
I'm not even sure what the py file is! I ran the patcher as it was, selected to have everything open and played. I'm uploading the video I made earlier and the rom to my onedrive.

**Comment by Trudovah** [comment]
Here is the link to the onedrive folder with both: https://1drv.ms/f/s!As2hDyU1pAGYb8iiuJWUdKJxmVg 
I hope this website lets me share it! lol

**Comment by flagrama** [comment]
I'm just curious, are you definitely using the 2.0 version of the randomizer or the 1.0 version?

**Comment by Trudovah** [comment]
Version 2.0.0.0. Just checked.

**Comment by Zelsaard** [comment]
I can confirm that this problem is resolved (at least for me) by using Python 3.6 or newer when creating the patched rom when using the 2.0 version of the randomizer.

**Comment by TopHatMan** [comment]
somehow the line that patches flags are not active. I think thats what causes it
`  write_bits_to_save(0x0EDC, 0x80) # "Entered the Master Sword Chamber"
    write_bits_to_save(0x0EDD, 0x20) # "Pulled Master Sword from Pedestal"`

These 2 by removing this or commenting it. You get the cs from pulling MS. Then Shiek comes and inf loops. 

Removing fariy ocarina gift skip does same thing. But you should not remove those.

**Comment by Trudovah** [comment]
So, if I inject these lines, it should work, correct?

**Comment by AmazingAmpharos** [comment]
Okay, I understand the issue here I think. Using a deprecated version of python caused a bit write in the save context not to work, and then everything makes sense. To explain what's going on under the hood here so you understand the situation...

The game has a variety of flags to determine when certain events have happened or not happened. We hook into the save file writing procedure to write various of these flags when a save file is initially created. Several of these flags we set write flags that tell the game that Link has been an adult before and talked to Sheik at the ToT before. We didn't actually remove the events per se; we actually just made the game believe the events had already occurred all of the time. Later we did commandeer Sheik's textboxes from that scene to form the new pedestal text that tells you about where the various medallions/stones are located. So, if you somehow get a save file in randomizer without the proper bits set (either because your version of python improperly runs the code or because you imported a save file from not randomizer), Sheik will use text boxes not properly built for her events anymore, and the behavior you saw will occur. An easy way to tell if a randomizer save is forming correctly is if you start the game with a randomly chosen medallion/stone and if there is no intro cutscene; if it's not that way, you do not have a randomizer save file, and a lot of bad things will happen to you that will almost assuredly include the game being impossible to complete (the stone/medallion you were "supposed to" start with won't exist anywhere else in the world!).

Thank you for the report, but I believe this is resolved.
